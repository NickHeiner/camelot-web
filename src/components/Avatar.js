import React from 'react';
import './Avatar.less';

export default ({currentUser}) => (
    currentUser ? 
        <img src={currentUser.photoURL} 
            alt={`avatar for ${currentUser.displayName}`} 
            className="profile-picture" />
        : <span className="profile-picture unknown" aria-label="No one">?</span>
);



