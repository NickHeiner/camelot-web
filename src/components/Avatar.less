.profile-picture {
  height: 60px;
  width: 60px;
  border-radius: 50%;

  &.unknown {
      background-color: #bbb;
      font-size: 36px;
      font-weight: bold;
      display: flex;
      justify-content: center;
      align-items: center;
  }

  &.active {
    border: 2px red solid;
  }
}
