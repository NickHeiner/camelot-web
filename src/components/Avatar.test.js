import Avatar from './Avatar';
import React from 'React';
import { shallow } from 'enzyme';

describe('Avatar', () => {
    it('renders with no user', () => {
        expect(shallow(
            <Avatar />
        ).html()).toMatchSnapshot();
    });

    describe('has a current user', () => {
        it('inactive', () => {
            const currentUser = {
                photoURL: 'photoURL',
                displayName: 'Display Name'
            };
            expect(shallow(
                <Avatar currentUser={currentUser} />
            ).html()).toMatchSnapshot();
        });

        it('active', () => {
            const currentUser = {
                photoURL: 'photoURL',
                displayName: 'Display Name'
            };
            expect(shallow(
                <Avatar currentUser={currentUser} isActive />
            ).html()).toMatchSnapshot();
        });
    });
});
