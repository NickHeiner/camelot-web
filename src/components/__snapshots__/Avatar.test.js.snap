exports[`Avatar has a current user active 1`] = `"<img src=\"photoURL\" alt=\"avatar for Display Name\" class=\"profile-picture active\"/>"`;

exports[`Avatar has a current user inactive 1`] = `"<img src=\"photoURL\" alt=\"avatar for Display Name\" class=\"profile-picture \"/>"`;

exports[`Avatar renders with no user 1`] = `"<span class=\"profile-picture unknown\" aria-label=\"No one\">?</span>"`;
